develop-guide/writeapi/writeapi.rst

```diff
@@ -0,0 +1,436 @@
+.. _write-api:
+
+#############
+Write an API
+#############
+
+.. important:: Before beginning development of a new API, be sure to raise an issue on GitHub letting the community know which API you will be creating. This allows us to avoid duplicate API development.
+
+.. note:: All APIs are contained within ``genielibs/src/sdk/apis``. This is where any new APIs should be added.
+
+APIs are the functional tools of Genie. They are designed to do one task and one task only, and should be as simple as
+possible. When making an API ask yourself: *what will this API do*? Configure a service? Get some information? Verify a 
+configuration has been set correctly? Whatever you decide, the API should only do a single task. This is by design, and helps to make APIs reusable. When building APIs, the goal is to have many small APIs that perform individual tasks rather than one single, large API that performs all necessary tasks.
+
+In these examples, we will be creating 3 types of APIs: one for configuring a device, one for getting information from a device, and one for verifying <something> on a device.
+
+These APIs will be placed in these files:
+
+.. code-block:: bash
+
+    genielibs/src/sdk/apis/iosxe/interface/configure.py
+    genielibs/src/sdk/apis/iosxe/interface/get.py
+    genielibs/src/sdk/apis/iosxe/interface/verify.py
+
+Configure
+"""""""""
+
+Configure APIs connect to the given device and execute commands to perform a task.
+
+.. tabs::
+
+    .. tab:: 1
+
+        .. code-block:: python
+
+            from unicon.core.errors import SubCommandFailure
+
+            def shut_interface(device, interface):
+                """ Shut interface
+            
+                    Args:
+                        device (`obj`): Device object
+                        interface (`str`): Interface name
+            
+                    Returns:
+                        None
+            
+                    Raises:
+                        SubCommandFailure
+                """
+
+        In this case, we want to create a configure API that will shutdown a given interface. To start we import SubCommandFailure to be used later, then define the function shut_interface. ``device`` is a required argument for all APIs, but since we also want to pass in an interface to shutdown, we include that as a parameter as well.
+        A docstring detailing what the API does, its arguments, its return value, and what exceptions it might raise is required. This will be used to populate the public API documenation.
```

Comment:
Would be nice to add something here about having to specify if an arg is optional and saying what the default value is
```
some_optional_arg(`str`, Optional): A thing. Default is None
```